### PR TITLE
Spike/ka 394 get rid of i delivery client internal dependencies

### PR DESCRIPTION
**Pull request by matus12**
Spike

### CODE REVIEW

File: KenticoCloud.Delivery.Tests/DependencyInjectionTests.cs [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
-using KenticoCloud.Delivery.InlineContentItems;
+﻿//using System;
+//using Microsoft.Extensions.DependencyInjection;
+//using Xunit;
+//using FakeItEasy;
+//using KenticoCloud.Delivery.InlineContentItems;
 
-namespace KenticoCloud.Delivery.Tests
-{
-    public class DependencyInjectionTests
```
Comment (petrsvihlik) [inline KenticoCloud.Delivery.Tests/DependencyInjectionTests.cs]: Finish this class :)

File: KenticoCloud.Delivery/DeliveryClientBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
+﻿using KenticoCloud.Delivery.InlineContentItems;
+using KenticoCloud.Delivery.ResiliencePolicy;
+using Microsoft.Extensions.Options;
+
+namespace KenticoCloud.Delivery
+{
+    public class DeliveryClientBuilder
```
[the remark this answers is not in the document]
Comment (petrsvihlik) [reply KenticoCloud.Delivery/DeliveryClientBuilder.cs]: also, add a method for adding a custom httpclient :)

File: KenticoCloud.Delivery/IServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            services.AddOptions();
+            services.TryAddTransient(serviceProvider => (IContentLinkUrlResolver) null);
+            services.TryAddTransient<IInlineContentItemsResolver<object>, ReplaceWithWarningAboutRegistrationResolver>();
+            services.TryAddTransient<IInlineContentItemsResolver<UnretrievedContentItem>, ReplaceWithWarningAboutUnretrievedItemResolver>();
+            services.TryAddTransient<IInlineContentItemsProcessor, InlineContentItemsProcessor>();
+            services.TryAddTransient<ICodeFirstModelProvider, CodeFirstModelProvider>();
+
+            services.TryAddTransient<IResiliencePolicyProvider>(serviceProvider =>
+            {
+                var deliveryOptions = serviceProvider.GetService<IOptions<DeliveryOptions>>();
```
[the remark this answers is not in the document]
Comment (matus12) [reply KenticoCloud.Delivery/IServiceCollectionExtensions.cs]: I have tested this and it worked.

**Comment by matus12** [comment]
The issue is being implemented in new [branch](https://github.com/Kentico/delivery-sdk-net/tree/DRAFT-4768_Get_rid_of_IDeliveryClient_internal_dependencies).

**Pull request closed without merging**